HHS/Head-Start-TTADP
# [TTAHUB- 2650] Fix session delete

## Description of change
The session deletion feature was broken when there was a goal associated with the event. In this case, the event/goal link table also included a session ID. This linkage generated a foreign key constraint error when attempting to delete a session. The fix involves allowing null values in the sessionId field of the link table, as we still need to retain the records there.


## How to test

- Make sure there are recipients of an event.
- Add a goal to the event. 
- Add a session to that event.
- Verify that the session can be deleted.
- Change the goal's text.
- Confirm that the goal was updated for the correct recipients.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2650


## Checklists

### Every PR

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Before merge to main


### Production Deploy


### After merge/deploy
